src/overlay/Floodgate.h

```diff
@@ -38,7 +38,7 @@ class Floodgate
 
         uint32_t mLedgerSeq;
         StellarMessage mMessage;
-        std::set<Peer::pointer> mPeersTold;
+        std::set<std::string> mPeersTold;
```

Comment:
If this is about restricting unnecessary retention, I think a weak_ptr ought to do. Bonus: you can clear the expired entries out of each flood record as you go.

Reply:
I thought about this but picked a unique ID instead:
* unique ID persist between re-connections, so we "remember" to not flood specific messages even if peers disconnect/reconnect (when using ptrs, we would not be able to do that)
* floodmap entries expire after a few ledgers, so I didn't think that we would clean those up more than what we do now